executor/containerdexecutor/executor_windows.go

```diff
@@ -88,6 +89,27 @@ func (w *containerdExecutor) createOCISpec(ctx context.Context, id, _, _ string,
 		return nil, nil, err
 	}
 	releasers = append(releasers, cleanup)
+
+	if v, ok := ctx.Value(appdefaults.ContextKeyCustomFrontend).(bool); ok && v {
+		sid := func() string {
+			for _, e := range meta.Env {
+				if strings.HasPrefix(e, "BUILDKIT_SESSION_ID=") {
+					return strings.TrimPrefix(e, "BUILDKIT_SESSION_ID=")
+				}
+			}
+			return ""
+		}()
+
+		frontendGrpcBridge := appdefaults.FrontendGRPCPipe + sid
```

Comment:
looking at this, seems `BUILDKIT_SESSION_ID` is passed to the frontend container meaning you can read it there as an ENV variable